### PR TITLE
Adds --cluster support

### PR DESCRIPTION
**Pull request by flovilmart**
Adds ability to start a mulitprocessed parse-server through cluster when using the CLI.

- Use `--cluster` to automatically select the number of processes
- Use `--cluster 2` to start 2 processes

### CODE REVIEW

**Comment by steven-supersolid** [comment]
When a forked process exits, the behaviour is a little different to throng which seems to be restarting that process: https://github.com/hunterloftis/throng/blob/master/lib/throng.js#L39 This would be important in keeping a constant number of processes when one of them has a problem.

LGTM apart from that and really useful.

**Comment by flovilmart** [comment]
Uhh you're right :) so call another fork on('exit'), I'll run some tests locally to see what's going on

File: src/cli/parse-server.js [hunk 1]
```diff
@@ -85,7 +85,8 @@ if (options.cluster) {
       cluster.fork();
     }
     cluster.on('exit', (worker, code, signal) => {
-      console.log(`worker ${worker.process.pid} died`);
+      console.log(`worker ${worker.process.pid} died... Restarting`);
+      cluster.fork();
```
Comment (flovilmart) [inline src/cli/parse-server.js]: @steven-supersolid I believe that will do :)

**Comment by facebook-github-bot-5** [comment]
@flovilmart updated the pull request - [view changes](https://github.com/ParsePlatform/parse-server/pull/2596/files/826a341cad76b088677c16b1ba00a8f20c5e3103..af5d49af0c6c1359ebf7d7fc44d3ff165aa632e1)

**Comment by flovilmart** [comment]
Merging as reviewed by @steven-supersolid

**Pull request merged**